log4j-core/src/main/java/org/apache/logging/log4j/core/impl/JdkMapAdapterStringMap.java

```diff
@@ -52,6 +52,11 @@ public JdkMapAdapterStringMap() {
 
     public JdkMapAdapterStringMap(final Map<String, String> map) {
         this.map = Objects.requireNonNull(map, "map");
+        try {
+            map.replace(Strings.EMPTY, Strings.EMPTY, Strings.EMPTY);
+        } catch (final UnsupportedOperationException ignored) {
+            immutable = true;
+        }
```

Comment:
Why don't we simply do `this.map = new HashMap<>(map)`? Checking against `Map#replace()` implies that `Map#replace()` is the only effect expected to be used and a future change to this contract will cause similar issues again.

Reply:
All the immutable maps in JDK (`Collections` and `ImmutableCollections`) I found throw on `Map#replace`.

I get your point, but I think the chances this issue resurfaces in the future, because an implementation allows `Map#replace` but does not implement other methods, is slim.
Adding `new HashMap` will incur in an allocation even if the map is already mutable.

Reply:
I won't object the change, but I wouldn't implement this myself. It is a bet for an unsolicited optimization.